platforms/android/library/src/main/java/io/element/android/wysiwyg/inputhandlers/InterceptInputConnection.kt

```diff
@@ -235,7 +243,13 @@ internal class InterceptInputConnection(
         // append to the current composition with the hardware keyboard.
         finishComposingText()
 
-        return setComposingText(newChars, 1)
+        return if (newChars.isDigitsOnly()) {
```

Comment:
This check has moved to `onHardwareCharacterKey()` - don't we need to handle the same case for software keyboards?

Reply:
I don't think so: after taking another look at the debugger, it turns out the problematic keyboards (Simple Keyboard, AnySoftKeyboard) just report the input events for these digits as hardware key presses, while Gboard uses `commitText`.